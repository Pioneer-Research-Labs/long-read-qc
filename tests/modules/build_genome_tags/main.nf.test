nextflow_process {

    name "Test Process build_genome_tags"
    script "modules/build_genome_tags/main.nf"
    process "build_genome_tags"

    test("Should generate genome tags") {

        when {
            params {
                 outdir = outputDir
            }
            process {
                """
                // define inputs of the process here
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/c.00432.dna'),
                file('${projectDir}/test_data/Tesseract_Barcode_Tracking_Grid.csv'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                "$outputDir/Gateway_EmptyBarcodedLib1_20241217/combined_tags.csv",
                "$outputDir/Gateway_EmptyBarcodedLib1_20241217/prefixed_tags.csv",
            ).match()
        }

    }

}
