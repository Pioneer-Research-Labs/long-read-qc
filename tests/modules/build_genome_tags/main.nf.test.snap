{
    "Should generate genome tags": {
        "content": [
            "combined_tags.csv:md5,02e69e651be84ddcfbb422dda08e3ee2",
            "prefixed_tags.csv:md5,e4eb395740c83f03c22776c0320d2932"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-18T21:05:29.302349"
    }
}